core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -228,6 +226,52 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+
+		pendingNonce, err := ec.getPendingNonce(ctx, from)
```

Comment:
Two things here:

1. This is going to make a _lot_ of requests to the eth node for pending nonce
2. We should use our own number in the database for pending nonce because chain is not reliable

Reply:
ok so i changed it to getLatestBlockNonce, which is basically eth_getTransactionCount count with nil argument.

But what do you mean about a lot of requests? We have usually just 2 'from' accounts in the node, so this should be up to 2 requests per each head received, no?

Reply:
Aaaah I misunderstood, I thought this was ranging over every attempt. My bad.

Reply:
ah no worries